svg.panzoom.js

```diff
@@ -129,16 +133,48 @@ SVG.extend(SVG.Doc, SVG.Nested, {
       lastP = currentP
     }
 
-    this.on('wheel', zoom)
+    this.on('wheel', wheelZoom)
     this.on('touchstart', pinchZoomStart, this, {passive:false})
     this.on('mousedown', panStart, this, {passive:false})
     this.on('touchstart', panStart, this, {passive:false})
 
     return this
 
-  }
+  },
+
+  zoom: function(level, point) {
+    var style = window.getComputedStyle(this.node)
+      , width = parseFloat(style.getPropertyValue('width'))
+      , height = parseFloat(style.getPropertyValue('height'))
+      , v = this.viewbox()
+      , zoomX = width / v.width
+      , zoomY = height / v.height
+      , zoom = Math.min(zoomX, zoomY)
 
+    if(level == null) {
+      return zoom
+    }
+
+    var zoomAmount = (zoom / level)
+
+    point = point || new SVG.Point(width/2 / zoomX + v.x, height/2 / zoomY + v.y)
+
+    var box = new SVG.Box(v)
+      .transform(new SVG.Matrix()
+        .scale(zoomAmount, point.x, point.y)
+      )
+
+    if(this.fire('zoom', {box: box, focus: point}).event().defaultPrevented)
+      return this
+
+    return this.viewbox(box)
+  },
 })
 
+SVG.extend(SVG.FX, {
+  zoom: function(level, point) {
+    return this.add('zoom', new SVG.Number(level), point)
```

Comment:
this feature was already pushed and will be available in the next release. It doenst hurt to include it here already. The old version just does nothing with the second parameter